### PR TITLE
MRG+1: interactivity in raw.plot_psd_topo() and epochs.plot_topo_image()

### LINKED ISSUE #6101: interactivity in epochs.plot_topo_image requires plt.ion()
It appears that the "click to magnify this channel" interactivity only works if the user has already imported pyplot and called `plt.ion()`. Try this in a fresh ipython session:

```
import os
import mne
sample_data_folder = mne.datasets.sample.data_path()
sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
                                    'sample_audvis_raw.fif')
raw = mne.io.read_raw_fif(sample_data_raw_file, verbose=False)
events = mne.find_events(raw, stim_channel='STI 014')
epochs = mne.Epochs(raw, events)
epochs.plot_topo_image()
# now click on any channel's imagemap, nothing happens
# now close the plot window, and...
import matplotlib.pyplot as plt
plt.show()  # only now does the enlarged image appear
```

Of course, if the last two steps aren't performed, then those orphaned figures hang around and only show up when the *next* MNE-Python plotting call is executed.  Would it work to use `matplotlib.is_interactive()` to get the ion/ioff state before the plot call, and if necessary turn it on for the duration of the plot, and then turn it back off again?

### PR DESCRIPTION
**Pull request by drammock**
closes #6101

- allows secondary figure popup when clicking in `epochs.plot_topo_image`, even if user hasn't set `plt.ion()`
- in `raw.plot_psd_topo`, avoids `TypeError: can only concatenate list (not "NoneType") to list` when clicking on a thumbnail sensor plot
- in popup figures spawned by `raw.plot_psd_topo` clicking:
    - avoids `TypeError: 'NoneType' object is not subscriptable` error on mouseover of axes area
    - fixes x axis label

### CODE REVIEW

File: mne/viz/topo.py [hunk 1]
```diff
@@ -426,6 +417,15 @@ def _rm_cursor(event):
     ax.xaxis.label.set_color(hvline_color)
     ax.yaxis.label.set_color(hvline_color)
 
+    if x_label is not None:
+        ax.set_xlabel(x_label)
```
Comment (drammock) [inline mne/viz/topo.py]: this needs to be done *after* `_setup_ax_spines` on line 410, which sets xlabel as `'Time (s)'`.  ylabel is not affected, but I thought it made sense to keep the `set_xlabel` and `set_ylabel` code together.

File: mne/viz/topo.py [hunk 2]
```diff
@@ -365,11 +357,11 @@ def _format_coord(x, y, labels, ax):
         timestr = '%6.3f s: ' % x
         if not nearby:
             return '%s Nothing here' % timestr
+        labels = [''] * len(nearby) if labels is None else labels
```
Comment (drammock) [inline mne/viz/topo.py]: moving this up from line 372, otherwise `labels[n]` on next line fails when `labels` is `None`

File: mne/viz/utils.py [hunk 3]
```diff
@@ -2332,6 +2332,7 @@ def _setup_ax_spines(axes, vlines, tmin, tmax, invert_y=False,
         axes.yaxis.set_label_coords(-0.05, 1 - ylabel_height
                                     if invert_y else ylabel_height)
 
+    vlines = [] if vlines is None else vlines
```
Comment (drammock) [inline mne/viz/utils.py]: could also make vlines parameter default to `[]` in the function signature, but that would have required reordering the function params.

File: mne/viz/topo.py [hunk 4]
```diff
@@ -347,44 +348,38 @@ def _plot_timeseries(ax, ch_idx, tmin, tmax, vmin, vmax, ylim, data, color,
...
         nearby_data = [(data[n], labels[n], times[n]) for n in nearby]
         ylabel = ax.get_ylabel()
-        unit = (ylabel[ylabel.find('(') + 1:ylabel.find(')')]
-                if '(' in ylabel and ')' in ylabel else '')
-        labels = [''] * len(nearby_data) if labels is None else labels
+        yunit = (ylabel[ylabel.find('(') + 1:ylabel.find(')')]
+                 if '(' in ylabel and ')' in ylabel else '')
```
Comment (drammock) [inline mne/viz/topo.py]: just changing variable name here from `unit` to `yunit` since now we also keep track of `xunit`

File: mne/viz/topo.py [hunk 5]
```diff
@@ -347,44 +348,38 @@ def _plot_timeseries(ax, ch_idx, tmin, tmax, vmin, vmax, ylim, data, color,
...
         """Create status string based on cursor coordinates."""
         # find indices for datasets near cursor (if any)
         tdiffs = [np.abs(tvec - x).min() for tvec in times]
         nearby = [k for k, tdiff in enumerate(tdiffs) if
                   tdiff < (tmax - tmin) / 100]
-        timestr = '%6.3f s: ' % x
+        xlabel = ax.get_xlabel()
+        xunit = (xlabel[xlabel.find('(') + 1:xlabel.find(')')]
+                 if '(' in xlabel and ')' in xlabel else 's')
+        timestr = '%6.3f %s: ' % (x, xunit)
```
Comment (drammock) [inline mne/viz/topo.py]: now this will correctly say "Hz" instead of "s" in the statusbar when hovering over a PSD plot.

**Comment by drammock** [comment]
@larsoner the addition of `plt.show()` is making the azure tests upset: `UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.` The specific test generating the warning is [here](https://github.com/mne-tools/mne-python/blob/master/mne/viz/tests/test_topo.py#L122-L125).  What is the right way to suppress / avoid this warning?

**Comment by larsoner** [comment]
Add a line to `setup.cfg` to ignore it

**Comment by drammock** [comment]
thanks @larsoner

**Pull request merged**